crates/dolos/src/client.rs

```diff
@@ -7,15 +9,18 @@ pub struct Dolos {
 }
 
 impl Dolos {
-    pub fn new(config: &Option<DolosConfig>) -> Result<Option<Self>, AppError> {
+    pub fn new(config: &Option<DolosConfig>) -> Result<Self, AppError> {
         if let Some(cfg) = config {
             let url = Url::parse(&cfg.endpoint).map_err(|e| AppError::Dolos(e.to_string()))?;
-            let client = JsonClient::new(url, cfg.request_timeout)?;
+            let client = JsonClient::new(url, cfg.request_timeout, false)?;
 
-            Ok(Some(Dolos { client }))
+            Ok(Dolos { client })
         } else {
-            // dolos endpoint not configured
-            Ok(None)
+            let is_disabled = true;
```

Comment:
Hmmm, now I'm thinking, if `JsonClient::is_disabled`, wouldn’t it be cleaner to have an `Option<JsonClient>` instead, without that field? Then you don't have to create fake URLs (that you’ll never use) etc.